c10/core/SymNodeImpl.h

```diff
@@ -164,16 +164,22 @@ class C10_API SymNodeImpl : public c10::intrusive_ptr_target {
   virtual std::string str() {
     TORCH_CHECK(false, "NYI");
   };
-  virtual int64_t large_negative_int() {
-    return 0; // not a large negative int!
+  virtual c10::optional<int64_t> singleton_int() {
+    return c10::nullopt;
+  }
+  virtual c10::optional<int64_t> constant_int() {
+    return c10::nullopt;
```

Comment:
This feels like XLA might need to implement it

Reply:
Oh wow didn't realize they have XLASymNodeImpl
Hmmm they didn't implement large_negative_number either though